/* Copyright (c) 2013 Stanford University
 *
 * Permission to use, copy, modify, and distribute this software for any
 * purpose with or without fee is hereby granted, provided that the above
 * copyright notice and this permission notice appear in all copies.
 *
 * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR(S) DISCLAIM ALL WARRANTIES
 * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
 * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL AUTHORS BE LIABLE FOR
 * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
 * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
 * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
 * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 */

syntax = "proto2";

package RAMCloud.ProtoBuf;

// This protocol buffer message is used to store persistent data needed
// by CoordinatorUpdateManager. These protocol buffers are used for
// storing and retrieving data from ExternalStorage.
message CoordinatorUpdateInfo {

    // All operations with sequence numbers less than or equal to this value
    // are known to have completed. After a coordinator crash, this tells the
    // new coordinator which operations it must check to ensure that they have
    // completed successfully.
    required uint64 lastFinished = 1;

    // Sequence numbers less than this value may have been assigned already,
    // so a new coordinator should not use values smaller than this. Used to
    // ensure that sequence numbers don't get reused after a coordinator crash.
    required uint64 firstAvailable = 2;
}
